### PR TITLE
various anchor related changes

### LINKED ISSUE #49: Copying links to doc pages sucks
1. When you go to a section, the #anchor-part-of-url isn't on the end in the progress bar; so it's hard to cite things on IRC.
2. readme.io also gave you a little anchor icon when you floated next to a link to get that anchor uri; would be nice for us to get that, too.

### PR DESCRIPTION
**Pull request by mlyle**
1. don't prevent the default handler for clicking an anchor, when doing
smooth scrolling.  This prevents fragments from being ignored / not
added to the address bar.
2. Include the excellent anchor.js, to add little 'anchor' links after
doc stuff.
3. Regenerate the javascript.

fixes #49

### CODE REVIEW

**Comment by mlyle** [comment]
image

**Comment by mlyle** [comment]
Need to make this only apply to docs pages, because it's a little wonky on the front page.

**Comment by tracernz** [comment]
Can we make the anchor a class and apply that class to the main content div (or whatever the content container) on the doc pages?

**Comment by mlyle** [comment]
Yes-- we need to do something like that.  The ones I've thought of are--

1. Have our own version of the anchorjs styling that suppresses it on most pages.
2. Figure out a way to not call the anchors stuff at all on non-doc pages.
3. Give anchors a selector that will only find things in the main body of a doc page.

File: assets/js/_main.js [hunk 1]
```diff
@@ -32,7 +32,9 @@ $(document).ready(function(){
   });
 
   // init smooth scroll
-  $("a").smoothScroll({offset: -20});
+  $("a").smoothScroll({offset: -20, preventDefault:false});
+
+  anchors.add();
```
Comment (tracernz) [inline assets/js/_main.js]: If we can add "doc" class to the headings in docs somehow, then:
```js
anchors.add('h2.doc, h3.doc, h4.doc, h5.doc, h6.doc')
```
It uses document.querySelectorAll if you pass a string: https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelectorAll

**Comment by tracernz** [comment]
Do you mind if I push a commit to this branch that restricts anchors to docs pages?

**Comment by tracernz** [comment]
P.S. how do you generate main.min.js? I just hand-edited it..

**Pull request merged**